### PR TITLE
Opengrok tools use pytest instead of unittest

### PR DESCRIPTION
**Pull request by tulinkry**
Consider using pytest as a default test library for our python tools.

### CODE REVIEW

File: opengrok-tools/pom.xml [hunk 1]
```diff
@@ -309,7 +311,7 @@ Portions Copyright (c) 2017-2018, Chris Fraire <cfraire@me.com>.
                                 <argument>setup.py</argument>
                                 <argument>test</argument>
                             </arguments>
-                            <skip>${skipTests}</skip>
+                            <skip>false</skip>
```
Comment (vladak) [inline opengrok-tools/pom.xml]: why this change ?

Comment (tulinkry) [reply opengrok-tools/pom.xml]: ah, this is leftover

**Comment by vladak** [comment]
Could you explain the rationale a bit ? The tests were duplicated because of a change to `setuptools` however this change requires certain version of that **and** switches to `pytest`. So, why do the latter ?

**Comment by tulinkry** [comment]
The problem with duplicated test is fixed by either of it. It’s still necessary to have higher version of setuptools for pytest 3+.

Choosing pytest over embedded unittest is a matter of personal choice supported by wide usage of pytest in other python project where I inspired by myself. Although you can achieve basically the same thing - testing the code, pytest offers less verbose code and is prepared for interaction with mocking libraries like flexmock or betamax which we don’t use (yet).

To sum it up, it doesn’t bring more than unittest at the moment, except for colours.

**Comment by tulinkry** [comment]
You can get more information about the differences for example here https://www.slant.co/versus/9148/9149/~unittest_vs_pytest
